src/core/gui/PageView.cpp

```diff
@@ -253,41 +243,40 @@ auto XojPageView::onButtonPressEvent(const PositionInputData& pos) -> bool {
              * This `if` statement can probably be replaced by an `assert` once #4377 is fixed
              */
             g_warning("InputHandler already exists upon XojPageView::onButtonPressEvent. Deleting it (and its views)");
-            eraseViewsOf(this->overlayViews, this->inputHandler);
-            delete this->inputHandler;
-            this->inputHandler = nullptr;
+            eraseViewsOf(this->overlayViews, this->inputHandler.get());
+            this->inputHandler.reset();
         }
 
         Control* control = this->xournal->getControl();
         switch (h->getDrawingType()) {
             case DRAWING_TYPE_LINE:
-                this->inputHandler = new RulerHandler(control, getPage());
+                this->inputHandler = (std::make_unique<RulerHandler>(control, getPage()));
```

Comment:
There is no need for extra brackets here and in the other cases.
```suggestion
                this->inputHandler = std::make_unique<RulerHandler>(control, getPage());
```